recipes/ffmpeg/all/conanfile.py

```diff
@@ -0,0 +1,472 @@
+        if self.options.get_safe("with_pulse"):
+            self.requires("pulseaudio/14.2")
+        if self.options.get_safe("with_vaapi"):
+            self.requires("vaapi/system")
+        if self.options.get_safe("with_vdpau"):
+            self.requires("vdpau/system")
+
+    def validate(self):
+        if self.options.with_ssl == "securetransport" and not tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("securetransport is only available on Apple")
+        if self.settings.os == "Macos" and self.settings.arch == "armv8" and self.options.with_libvpx:
+            raise ConanInvalidConfiguration("libvpx doesn't support armv8 (yet)")
+        if self.settings.os in ["Linux", "FreeBSD"] and self.options.with_sdl:
+            raise ConanInvalidConfiguration("sdl is not supported by this recipe (yet)")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _arch(self):
+        return {
+            "armv6": "arm",
+            "armv7": "arm",
+            "armv8": "aarch64",
+            "x86": "x86",
+            "x86_64": "x86_64",
+        }.get(str(self.settings.arch))
+
+    def _patch_sources(self):
+        if self.settings.compiler == "Visual Studio" and self.options.with_libx264 and not self.options["libx264"].shared:
+            # suppress MSVC linker warnings: https://trac.ffmpeg.org/ticket/7396
+            # warning LNK4049: locally defined symbol x264_levels imported
+            # warning LNK4049: locally defined symbol x264_bit_depth imported
+            tools.replace_in_file(os.path.join(self._source_subfolder, "libavcodec", "libx264.c"),
+                                  "#define X264_API_IMPORTS 1", "")
+        if self.options.with_ssl == "openssl":
+            # https://trac.ffmpeg.org/ticket/5675
+            openssl_libraries = " ".join(["-l%s" % lib for lib in self.deps_cpp_info["openssl"].libs])
+            tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                                  "check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||",
+                                  "check_lib openssl openssl/ssl.h OPENSSL_init_ssl %s || " % openssl_libraries)
+
+    @contextlib.contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self):
+                yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        self._autotools.libs = []
+        opt_enable_disable = lambda what, v: "--{}-{}".format("enable" if v else "disable", what)
+        args = [
+            "--pkg-config-flags=--static",  # FIXME: needed?
```

Comment:
it's here from the very beginning - was added according to [the official compilation guide](https://trac.ffmpeg.org/wiki/CompilationGuide/Ubuntu)